build_tools/circle/build_doc.sh

```diff
@@ -183,7 +184,7 @@ ccache -s
 
 export OMP_NUM_THREADS=1
 
-if [[ "$CIRCLE_BRANCH" =~ ^main$ || -n "$CI_PULL_REQUEST" ]]
+if [[ "$CIRCLE_BRANCH" =~ ^main$ || ( -n "$CI_PULL_REQUEST" && "$CIRCLE_TARGET_BRANCH" =~ ^main$ ) ]]
```

Comment:
you named it ``CI_TARGET_BRANCH`` above. Why is it not ``CIRCLE_BRANCH`` ?

Reply:
Because we are running in GitHub Action nowadays :)

Reply:
The script is translating the variables

Reply:
can you point me where ? I can't find it

Reply:
Uhm Actually I'm wrong. It seems that our code work on both CircleCI and GitHub Actions but we are still using CircleCI.

https://github.com/scikit-learn/scikit-learn/blob/9012b787bdded56c7d189043b88f8dfc0dbe911a/build_tools/circle/build_doc.sh#L24-L27

So it means that my fix is not working. I should instead find the corresponding CircleCI environment variable.

Reply:
Honestly we should remove the Github translation code. There may be a little bit of hope that one day Github allows to have PR previews for example https://github.com/actions/deploy-pages/issues/337 and https://github.com/orgs/community/discussions/7730#discussioncomment-1885967 but in the foreseeable future I think that the doc build is going to stay on CircleCI ...

Context was that at one point we had quite a complicated setup to build on Github actions, upload an artifact, download the artifact on CircleCI and have the rendered doc on CircleCI, see https://github.com/scikit-learn/scikit-learn/pull/25466.